diff --git a/src/stripe.js b/src/stripe.js
index 53e417b..f6ff9a8 100644
--- a/src/stripe.js
+++ b/src/stripe.js
@@ -8,6 +8,7 @@ const logger = require('./logger');
 const stripe = new Stripe(config.stripe.secret_key);
 
 async function handleStripeEvent(event) {
+	return;
 	if (event.type === 'customer.subscription.updated' || event.type === 'customer.subscription.deleted') {
 		const subscription = event.data.object;
 		const product = await stripe.products.retrieve(subscription.plan.product);
