diff --git a/Dockerfile b/Dockerfile
index 16a518b..5045dc6 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -9,6 +9,11 @@ ARG app_dir
 
 WORKDIR ${app_dir}
 
+RUN apk add git
+RUN git clone --depth 1 --branch v1.0.22 https://github.com/PretendoNetwork/nex-protocols-common-go
+# Terrible hack to work around the issue where the ticket timestamp is 0
+RUN sed -i 's/ticket\.Timestamp()\.Standard()/time\.Now()\.UTC()/g' nex-protocols-common-go/secure-connection/connect.go
+
 RUN --mount=type=cache,target=/go/pkg/mod/ \
 	--mount=type=bind,source=go.sum,target=go.sum \
 	--mount=type=bind,source=go.mod,target=go.mod \
diff --git a/go.mod b/go.mod
index 2dd4e82..1bb70c2 100644
--- a/go.mod
+++ b/go.mod
@@ -4,6 +4,8 @@ go 1.23.0
 
 toolchain go1.23.4
 
+replace github.com/PretendoNetwork/nex-protocols-common-go => ./nex-protocols-common-go
+
 require (
 	github.com/PretendoNetwork/grpc-go v1.0.2
 	github.com/PretendoNetwork/nex-go/v2 v2.1.2
